### PR TITLE
Cyclomatic complexity

### PR DESCRIPTION
**Pull request by kinegratii**
This PR add cyclomatic complexity for project.

Usage

```shell
$  flake8 pyecharts --max-complexity 10
pyecharts\echarts\option.py:430:1: C901 '_mark' is too complex (11)
pyecharts\echarts\option.py:639:1: C901 'visual_map' is too complex (12)
```

### CODE REVIEW

File: pyecharts/utils/__init__.py [hunk 1]
```diff
@@ -4,6 +4,9 @@
 import codecs
 import os
 
+__all__ = ['get_resource_dir', 'write_utf8_html_file', 'to_css_length',
+           'merge_js_dependencies']
```
Comment (chfw) [inline pyecharts/utils/__init__.py]: with this, you could allow `_add` to be outside merge_js_dependencies.

File: pyecharts/utils/__init__.py [hunk 2]
```diff
@@ -45,16 +45,16 @@ def to_css_length(x):
...
-    if hasattr(ele, 'js_dependencies'):
-        return list(ele.js_dependencies)
-    if isinstance(ele, (list, tuple, set)):
-        return ele
-    return ele,
+    if hasattr(obj, 'js_dependencies'):
+        return list(obj.js_dependencies)
+    if isinstance(obj, (list, tuple, set)):
+        return obj
+    return obj,  # tuple
```
Comment (chfw) [inline pyecharts/utils/__init__.py]: the last turn means we are returning it anyway. should you raise an exception here? tuple is detected in previous if. with the exception, the developer knows it during the development.

File: pyecharts/utils/__init__.py (hunk 1 above)
Comment (kinegratii) [reply pyecharts/utils/__init__.py]: `_add` references `dependencies` and `fist_items` by pointer (Not value reference)

File: pyecharts/utils/__init__.py (hunk 2 above)
Comment (kinegratii) [reply pyecharts/utils/__init__.py]: `_flat` and `merge_js_dependencies` only accepts chart's dependencies as their input,which dependency building process should ensure.

**Pull request merged**